tests/testthat/test-partial-eval.R

```diff
@@ -93,6 +169,128 @@ test_that("old _at functions continue to work", {
   expect_snapshot(lf %>% dplyr::summarise_at(dplyr::vars(a:b), ~ sum(.)))
 })
 
+test_that("across() defaults to everything()", {
+  # SELECT `x` + 1.0 AS `x`, `y` + 1.0 AS `y`
+  expect_snapshot(
+    lazy_frame(x = 1, y = 1) %>% summarise(across(.fns = ~ . + 1))
+  )
+})
+
+test_that("across() passes ... to functions", {
+  expect_equal(
+    partial_eval_dots(quos(across(everything(), log, base = 2)), c("a", "b")),
+    unclass(quos(a = log(a, base = 2), b = log(b, base = 2)))
+  )
+
+  expect_equal(
+    partial_eval_dots(
+      quos(across(everything(), list(times = `*`, plus = `+`), 2)),
+      vars = c("a", "b")
+    ),
+    quos(
+      a_times = a * 2,
+      a_plus = a + 2,
+      b_times = b * 2,
+      b_plus = b + 2
+    ) %>%
+      unclass()
+  )
+})
+
+test_that("across(.names=) can use local variables in addition to {col} and {fn}", {
+  res <- local({
+    prefix <- "MEAN"
+    lazy_frame(x = 42) %>%
+      summarise(across(everything(), ~ mean(.x, na.rm = TRUE), .names = "{prefix}_{.col}"))
+  })
+  expect_equal(op_vars(res), "MEAN_x")
+})
+
+test_that("across() uses environment from the current quosure (dplyr#5460)", {
+  # If the data frame `y` is selected, causes a subscript conversion
+  # error since it is fractional
+  df <- lazy_frame(x = 1, y = 2.4)
+  y <- "x"
+
+  expect_equal(
+    partial_eval_dots(quos(across(all_of(y), mean)), c("x", "y")),
+    list(x = quo(mean(x)))
+  )
+
+  expect_equal(
+    partial_eval_dots(quos(if_all(all_of(y), ~ .x < 2)), c("x", "y")),
+    set_names(list(quo(x < 2)), "")
+  )
+})
+
+test_that("lambdas in across() can use columns", {
+  expect_equal(
+    partial_eval_dots(quos(across(everything(), ~ .x / y)), c("x", "y", "z")),
+    list(
+      x = quo(x / y),
+      y = quo(y / y),
+      z = quo(z / y)
+    )
+  )
+
+  skip(".data not yet supported in across()")
+  df <- tibble(x = 2, y = 4, z = 8)
+  lf <- lazy_frame(x = 2, y = 4, z = 8)
+  expect_identical(
+    df %>% mutate(data.frame(x = x / y, y = y / y, z = z / y)),
+    df %>% mutate(across(everything(), ~ .x / .data$y))
+  )
+})
+
+test_that("can pass quosure through `across()`", {
+  summarise_mean <- function(data, vars) {
+    data %>% summarise(across({{ vars }}, ~ mean(.x, na.rm = TRUE)))
+  }
+  gdf <- lazy_frame(g = c(1, 1, 2), x = 1:3) %>% group_by(g)
+
+  expect_equal(
+    gdf %>% summarise_mean(x) %>% remote_query(),
+    summarise(gdf, x = mean(x, na.rm = TRUE)) %>% remote_query()
+  )
+})
+
+test_that("across() can use named selections", {
+  df <- lazy_frame(x = 1, y = 2)
+
+  # no fns
+  expect_equal(
+    df %>% summarise(across(c(a = x, b = y))) %>% remote_query(),
+    sql("SELECT `x` AS `a`, `y` AS `b`\nFROM `df`")
```

Comment:
I used `expect_equal()` because the queries are very simple. Alternatively, this could use `op_vars()` though this does not fully capture what's going on. What do you think?